### PR TITLE
Loki: Show loading state longer when running query

### PR DESCRIPTION
**Pull request by zoltanbedi**
**What this PR does / why we need it**:

@slim-bean raised the issue that the spinner in the right corner is only showing for a short period of time but the query to show up takes longer.

### CODE REVIEW

File: public/app/plugins/datasource/loki/datasource.ts [hunk 1]
```diff
@@ -95,7 +95,12 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
 
     filteredTargets.forEach(target =>
       subQueries.push(
-        this.runInstantQuery(target, options, filteredTargets.length),
+        this.runInstantQuery(target, options, filteredTargets.length).pipe(
```
Comment (aocenas) [inline public/app/plugins/datasource/loki/datasource.ts:98]: Hmm I think this should be done in Explore itself. First there is no guaranties that instant query will finish first. Also the problem seems to be generic in the sense that the same issue may be in other datasources too and with other combinations of queries.

Explore should have something like `loading = queries.some(q => x.isLoading)`.

Comment (torkelo) [reply public/app/plugins/datasource/loki/datasource.ts:98]: in the dashboard to optimize performance and not render partial results as every query completes we wait to render until we get a Done.

if data source does not return a state we set state to Done automatically after first result:
https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/state/runRequest.ts#L44 

it's a bit tricky for Grafana to know what state should be as some queries do not return any results, and some queries trigger multiple requests and results. So not sure where this logic is best placed

Comment (torkelo) [reply public/app/plugins/datasource/loki/datasource.ts:98]: think in Prometheus we have different logic for queries from explore and dashboard, as in explore we return results as they complete and in dashboard Prometheus will wait for all queries to complete

File: public/app/plugins/datasource/loki/datasource.ts [hunk 2]
```diff
@@ -98,7 +98,14 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
 
     for (const target of filteredTargets) {
       if (options.app === CoreApp.Explore) {
-        subQueries.push(this.runInstantQuery(target, options, filteredTargets.length));
+        subQueries.push(
+          this.runInstantQuery(target, options, filteredTargets.length).pipe(
+            map(response => {
+              // Let rangeQuery finish the loading
+              return { ...response, state: LoadingState.Loading };
```
Comment (aocenas) [inline public/app/plugins/datasource/loki/datasource.ts:105]: What if instant query returns after range query? Would explore be stuck in loading state?

Comment (zoltanbedi) [reply public/app/plugins/datasource/loki/datasource.ts:105]: You are right it would.

**Comment by zoltanbedi** [comment]
Closing this as this is not a proper fix for #28718

**Pull request closed without merging**